cocotb/types/array.py

```diff
@@ -128,59 +233,18 @@ class Array(Sequence):
         "_range",
     )
 
-    _value: Sequence
-    """
-    Private interface for subclasses to access the value as a :class:`~collections.abc.Sequence`
-
-    Subclasses that don't use a Sequence as their main representation should emulate this object,
-    or override *all* :class:`~cocotb.types.Array` methods *except*:
-        - :attr:`left`
-        - :attr:`direction`
-        - :attr:`right`
-        - :attr:`range`
-        - :attr:`__len__`
-        - :attr:`_translate_index`
-    """
-
-    def __init__(
-        self, value: Optional[Iterable[Any]] = None, range: Optional[Range] = None
-    ):
-        if value is not None and range is None:
-            self._value = list(value)
+    def __init__(self, value: typing.Iterable[T], range: typing.Optional[Range] = None):
+        self._value = list(value)
+        if range is None:
             self._range = Range(0, "to", len(self._value) - 1)
-        elif value is not None and range is not None:
-            if not isinstance(range, Range):
-                raise TypeError("range argument must be of type 'Range'")
-            self._value = list(value)
+        else:
             self._range = range
             if len(self._value) != len(self._range):
                 raise ValueError(
                     "init value of length {!r} does not fit in {!r}".format(
                         len(self._value), self._range
                     )
                 )
-        elif value is None and range is not None:
-            if not isinstance(range, Range):
-                raise TypeError("range argument must be of type 'Range'")
-            self._value = [None] * len(range)
```

Comment:
I assume we decided this was not useful?

Reply:
Yeah, now that it's a homogenous type you need a default value, instead of trying to figure out how to provide that I just removed the feature.